### PR TITLE
feat(trading,governance,wallet): browser wallet integration

### LINKED ISSUE #3819: Connect dapp to browser wallet
## Story

As a user
I want to be able to use the browser wallet beta release with the Console and Governance dapps to approve / reject transactions
So that I can experience it's awesomeness AND help Vega test and improve it 

## Acceptance Criteria

There are A/Cs already written and covered for Connect Wallet, Disconnect Wallet and Select and Switch Keys should pass against the browser wallet https://github.com/vegaprotocol/specs/blob/master/user-interface/0002-WCON-connect_vega_wallet.md

These may also be useful to consider if not already captured in the A/Cs above:
- [ ] I can use `client.connect_wallet` on the Vega API
- [ ] I can list permitted keys with `client.list_keys` after permissions are granted
- [ ] If I have no permissions for a given key `client.send_transaction` should fail
- [ ] If I have no permissions at all all `client.*` endpoints should fail except `client.connect_wallet`
- [ ] I can call `client.get_chain_id` without permissions
- [ ] I can receive and understand a message I receive from the browser wallet dapp telling me that it has disconnected from Console / Governance  

## Tasks

This ticket is dependent on https://github.com/vegaprotocol/vegawallet-browser/issues/211 as the browser wallet team would like the Console / Governance teams to be able to complete the integration using only the browser wallet documentation (not having any direct help from the team) in order to verify it is possible to do self serve.

Also, the browser wallet option may need to be live in fairground for ~6-8 weeks before being released to mainnet

- [ ] Read the documentation provided by the Browser Wallet team
- [ ] Detailed sketch (to incorporate the addition of another wallet option to the connect flow)
- [ ] Visual design (if needed)
- [ ] Implement UI tweaks and integration
- [ ] Write unit tests
- [ ] Write end to end tests
- [ ] Code review
- [ ] Team verify acceptance criteria met
- [ ] Exploratory QA testing 
- [ ] Capture bugs and new edge cases in tickets (this one or new TBD) 
- [ ] Write automated regression tests for the A/Cs (expanding on A/Cs where possible) 
- [ ] Add story and A/Cs to spec doc 
- [ ] Review with stakeholders 
- [ ] Release

## Notes

Speak to Dex who can guide you through the integration

Documentation https://github.com/vegaprotocol/documentation/pull/629

Deploy previews of browser extension stuff

https://deploy-preview-629--vega-docusaurus.netlify.app/testnet/api/vega-wallet/reference/browser-extension/get-started

https://deploy-preview-629--vega-docusaurus.netlify.app/testnet/api/vega-wallet/reference/browser-extension/javascript-api

### PR DESCRIPTION
**Pull request by mattrussell36**
# Related issues 🔗

Closes #3819 

# Description ℹ️

Adds support for the browser wallet

# Technical 👨‍🔧

- Updates injected connector with calls to window.vega
- Displays web wallet connection option in connect dialog if window.vega is detected
- Tidies up look and feel of connect dialog (icons, dialog size)

### CODE REVIEW

File: libs/wallet/src/connectors/injected-connector.ts [hunk 1]
```diff
@@ -1,21 +1,85 @@
-import type { VegaConnector } from './vega-connector';
+import { clearConfig, setConfig } from '../storage';
+import type { Transaction, VegaConnector } from './vega-connector';
+
+declare global {
+  interface Vega {
+    getChainId: () => Promise<{ chainID: string }>;
+    connectWallet: (params: { hostname: string }) => Promise<null>;
```
Comment (emilbayes) [inline libs/wallet/src/connectors/injected-connector.ts:7]: I believe the `hostname` param is gone from the desktop api? At least the web wallet does not take any arguments on `connectWallet`

File: libs/wallet/src/use-injected-connector.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+export const useInjectedConnector = (onConnect: () => void) => {
+  const { connect, acknowledgeNeeded } = useVegaWallet();
+  const [status, setStatus] = useState(Status.Idle);
+  const [error, setError] = useState<Error | null>(null);
+
+  const attemptConnect = useCallback(
+    async (connector: InjectedConnector, appChainId: string) => {
+      try {
+        console.log('here');
```
Comment (emilbayes) [inline libs/wallet/src/use-injected-connector.ts:22]: Stay console.log

**Review by macqbat — APPROVED** [review]

File: libs/wallet/src/connect-dialog/connect-dialog.tsx [hunk 3]
```diff
@@ -216,136 +190,129 @@ const ConnectorList = ({
...
           </li>
-        </ul>
-      </ConnectDialogContent>
-      <ConnectDialogFooter />
+        )}
+        <li className="mb-4 last:mb-0">
+          <div className="my-4 text-center">{t('OR')}</div>
+          <ConnectionOption
+            type="view"
+            text={t('View as vega user')}
```
Comment (sam-keen) [inline libs/wallet/src/connect-dialog/connect-dialog.tsx:232]: Suggestion (non-blocking): capitalised 'Vega'
